src/Raven.Server/Rachis/Follower.cs

```diff
@@ -173,6 +173,13 @@ private void NegotiateWithLeader(TransactionOperationContext context, AppendEntr
                     _engine.SetLastCommitIndex(context, snapshot.LastIncludedIndex, snapshot.LastIncludedTerm);
                     _engine.TruncateLogBefore(context, snapshot.LastIncludedIndex);
                 }
+                else
+                {
+                    if (_engine.GetLastEntryIndex(context) != snapshot.LastIncludedIndex)
+                    {
+                        throw new InvalidOperationException("The snapshot installation had failed we will need to get a new one...");
```

Comment:
Need to add actual semantics here, that is, why did it fail, including terms and index values.